actions/action.go

```diff
@@ -62,8 +65,27 @@ func NewOutputs(uri string, latestVersion *semver.Version, additionalOutputs Out
 	return outputs, nil
 }
 
-func (o Outputs) Write(writer io.Writer) {
+func (o Outputs) Write() {
+	outputFileName, ok := os.LookupEnv("GITHUB_OUTPUT")
+	if !ok {
+		panic(errors.New("GITHUB_OUTPUT is not set, see https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter"))
+	}
+	writer, err := os.OpenFile(outputFileName, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0666)
+	if err != nil {
+		panic(err)
+	}
+	defer writer.Close()
+	delimiter := generateDelimiter()
 	for k, v := range o {
-		_, _ = fmt.Fprintf(writer, "::set-output name=%s::%s\n", k, v)
+		_, _ = fmt.Fprintf(writer, "%s<<%s\n%s\n%s\n", k, delimiter, v, delimiter) // see https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings
+	}
+}
```

Comment:
Do you think there would be any benefit to only using the multiline approach for values that actually contain multiple lines? Off the top of my head, it would be less complex (no delimiter required). Most of the values are not going to contain multiple lines.

If we search for `\n` in the value and find something, then we could safely use `%s=%s`.

Reply:
I also thought about this, but as it would complicate the implementation without any real benefit I dropped the idea.

Maybe if you want to look at the resulting file manually, then it would probably be easier to read (but maybe mixing two different styles for multiline and singleline actually makes matters worse). Anyway, I don't think it is an use-case here.

Reply:
Fair enough. I don't have strong feelings on the subject. I'm fine leaving it this way.